src/utils/AutoDiscoveryUtils.js

```diff
@@ -33,6 +34,8 @@ export class ValidatedServerConfig {
     isUrl: string;
 
     isDefault: boolean;
+
+    warning: string;
```

Comment:
ideally we don't put state flags here - this is just a configuration blob, not an object meant to carry state to other components. 

`hsNameIsDifferent` arguably shouldn't be here either, but its role is slightly different.

Reply:
I agree it's not great, but I also don't want throw and change control flow of callers... Should validation methods return multiple objects then to contain the warning...?

Reply:
I think this is fine for now, just highlighting that the original design of this didn't really anticipate this kind of flag being required.